eland/operations.py

```diff
@@ -1233,58 +1233,55 @@ def _es_results(
+        # Only return requested field_names and add them to body
         _source = query_compiler.get_field_names(include_scripted_fields=False)
-        if _source:
-            # For query_compiler._client.search we could add _source
-            # as a parameter, or add this value in body.
-            #
-            # If _source is a parameter it is encoded into to the url.
-            #
-            # If _source is a large number of fields (1000+) then this can result in an
-            # extremely long url and a `too_long_frame_exception`. Therefore, add
-            # _source to the body rather than as a _source parameter
-            body["_source"] = _source
-        else:
-            body["_source"] = False
-
-        es_results: Any = None
-
-        # If size=None use scan not search - then post sort results when in df
-        # If size>10000 use scan
-        is_scan = False
-        if size is not None and size <= DEFAULT_ES_MAX_RESULT_WINDOW:
-            if size > 0:
-                es_results = query_compiler._client.search(
-                    index=query_compiler._index_pattern,
-                    size=size,
-                    sort=sort_params,
-                    body=body,
-                )
-        else:
-            is_scan = True
-            es_results = scan(
-                client=query_compiler._client,
-                index=query_compiler._index_pattern,
-                query=body,
-            )
-            # create post sort
-            if sort_params is not None:
-                post_processing.append(SortFieldAction(sort_params))
-
-        if is_scan:
-            while True:
-                partial_result, df = query_compiler._es_results_to_pandas(
-                    es_results, collector.batch_size(), collector.show_progress
-                )
-                df = self._apply_df_post_processing(df, post_processing)
-                collector.collect(df)
-                if not partial_result:
-                    break
+        body["_source"] = _source if _source else False
+
+        is_pagination_required: bool = False
+
+        if result_size and (0 <= result_size <= DEFAULT_ES_MAX_RESULT_WINDOW):
+            body["size"] = result_size
         else:
-            partial_result, df = query_compiler._es_results_to_pandas(es_results)
-            df = self._apply_df_post_processing(df, post_processing)
-            collector.collect(df)
+            body["size"] = DEFAULT_ES_MAX_RESULT_WINDOW
+            is_pagination_required = True
+
+        if sort_params is not None:
+            body["sort"] = [sort_params]
+        elif is_pagination_required:
+            body["sort"] = [{"_doc": "asc"}]
+
+        # To Improve performance
+        body["track_total_hits"] = False
+
+        # Initialize Point in time API
+        pit_id: str = query_compiler._client.open_point_in_time(
+            index=query_compiler._index_pattern, keep_alive="1m"
```

Comment:
Let's define `keep_alive` in one place and use everywhere, and let's set it to something larger like 5m since we're actively cleaning it up at the end of our query anyways.

Reply:
So, I am keeping it as "3m" because. ES Documentation says following:
<html>
<body>
<!--StartFragment-->

The keep_alive parameter tells Elasticsearch how long it should extend the time to live of the point in time.
--


<!--EndFragment-->
</body>
</html>

So, for every search it extends the time. :)

Reply:
Works for me :)